### PR TITLE
🐛 Fix "Cannot read property HPAY of null"

### PR DESCRIPTION
**Pull request by Jurollet**
Fix https://october.atlassian.net/browse/POF-944

Provides a safer way to extract a single transaction or a transaction list from SOAP Lemonway results.

Test : 
- add LW credentials from your local `.env` of the API to `test/index.js`
- in `test/get-money-in.js`, add the following in first test
```
  const start = moment().subtract(1, 'day').toDate();
  const end = moment().toDate();

    lemonway.Transaction.getMoneyInTransDetails(chance.ip(), {
        from: start,
        to: end,
      })
      .then(function (transactions) {
        return done();
      }).catch(done);
```
- without the fix, the test will crash with `Cannot read property HPAY of null`
- with the fix, the test will pass

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -557,7 +589,7 @@ Client.prototype.getMoneyInChequeDetails  = function (ip, _opts) {
     updateDate: utils.dateToUnix(_opts.from) || 0
   });
 
-  return this._request('GetMoneyInChequeDetails', ip, opts).get('TRANS').get('HPAY');
+  return this._request('GetMoneyInChequeDetails', ip, opts).then(this._extractTransactionList);
```
Comment (ChristopheTiet) [inline lib/client.js:592]:
```suggestion
  return this._request('GetMoneyInChequeDetails', ip, opts).then(this._extractTransaction);
```

**Review by ChristopheTiet — CHANGES_REQUESTED** [review]
Could not find any reference for `FastPay` or `MoneyInNeosurf`

Just a small mistake on `GetMoneyInChequeDetails`.

Did the test in real case. Thansk for the work.

File: lib/client.js (hunk 1 above)
Comment (Jurollet) [reply lib/client.js:592]: + check method `getMoneyInChequeDetails` in `transaction-factory` and you'll see it using a `.map` method

Comment (ChristopheTiet) [reply lib/client.js:592]: My bad.

**Review by ChristopheTiet — APPROVED** [review]

File: lib/client.js [hunk 2]
```diff
@@ -115,13 +115,45 @@ Client.prototype._request = function (method, ip, args, resultName) {
...
+  }
+
+  const value = transactions['HPAY'];
+
+  if (value === null || value === undefined) {
+    return [];
+  }
+
+  return value;
+}
```
Comment (Deprasos) [inline lib/client.js:148]: Is it really mandatory to use "Client" methods? 🤔 
```suggestion
function _extractTransaction(result) {
  const transaction = result['TRANS'];

  if (transaction === null || transaction === undefined) {
    return {};
  }

  const value = transaction['HPAY'];

  if (value === null || value === undefined) {
    return {};
  }

  return value;
}

function _extractTransactionList(results) {
  const transactions = results['TRANS'];

  if (transactions === null || transactions === undefined) {
    return [];
  }

  const value = transactions['HPAY'];

  if (value === null || value === undefined) {
    return [];
  }

  return value;
}
```

**Review by Deprasos — CHANGES_REQUESTED** [review]
Thanks for the PR, some comments 🙈

File: lib/client.js (hunk 2 above)
Comment (ChristopheTiet) [reply lib/client.js:148]: I think you can do the changes ^^.

Comment (Deprasos) [reply lib/client.js:148]: Yeah I will do them, but I've others review to do before 🙈

**Review by Deprasos — APPROVED** [review]
Thanks, tested in real case ok 🔥

**Pull request merged**